pkg/yurthub/network/dummyif_linux.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+	"net"
+
+	"k8s.io/klog"
+
+	"github.com/vishvananda/netlink"
+)
+
+type DummyInterfaceController interface {
+	EnsureDummyInterface(ifName string, ifIP net.IP) error
+	DeleteDummyInterface(ifName string) error
+	ListDummyInterface(ifName string) ([]net.IP, error)
+}
+
+type dummyInterfaceController struct {
+	netlink.Handle
+}
+
+// NewDummyInterfaceManager returns an instance for create/delete dummy net interface
+func NewDummyInterfaceController() DummyInterfaceController {
+	return &dummyInterfaceController{
+		Handle: netlink.Handle{},
+	}
+}
+
+// EnsureDummyInterface make sure the dummy net interface with specified name and ip exist
+func (dic *dummyInterfaceController) EnsureDummyInterface(ifName string, ifIP net.IP) error {
+	link, err := dic.LinkByName(ifName)
+	if err == nil {
+		addrs, err := dic.AddrList(link, 0)
+		if err != nil {
+			return err
+		}
+
+		for _, addr := range addrs {
+			if addr.IP != nil && addr.IP.Equal(ifIP) {
+				return nil
+			}
+		}
+
+		klog.Infof("ip address for %s interface changed to %s", ifName, ifIP.String())
+		return dic.AddrReplace(link, &netlink.Addr{IPNet: netlink.NewIPNet(ifIP)})
+	}
```

Comment:
if err != nil, why not returning err but calling addDummyInterface?  Maybe a comment is needed.

Reply:
yeah, add error handle here, if err is `Link not found` ,  we will call addDummyInterface, otherwise return err.